string color
int32 pixel_x
int32 pixel_y
float32 area